src/pages/dar_application/ResearcherInfo_new.js

```diff
@@ -174,8 +213,26 @@ export default function ResearcherInfo(props) {
             type: 'button', // default button element type inside a form is "submit".
             className: 'button button-white btn-xs',
             style: { marginTop: 25 },
-            onClick: () => {}
-          }, ['Add Collaborator'])
+            onClick: () => addNewCollaborator(),
+            onChange: ({key: name, value,}) => formFieldChange({name, value}),
+          }, [
+            span({}, ['Add Collaborator'])
+          ]),
+
+          collaboratorsState
+            .map((state, index) => {
+              if (isNil(state)) {
+                return div({}, []);
+              }
+
+              return h(CollaboratorForm, {
+                key: index,
```

Comment:
You should pass the state/collaborator data in as well to allow the form to initialize the data with the currently saved values.